mlir/lib/Dialect/Linalg/Transforms/Transforms.cpp

```diff
@@ -295,7 +296,7 @@ FailureOr<LowerPackResult> linalg::lowerPack(RewriterBase &rewriter,
       llvm::interleaveComma(stripMinedShape, DBGS() << "stripMinedShape: ");
       DBGSNL(); DBGS() << "collapsed type: " << collapsed; DBGSNL(););
 
-  if (packOp.isLikePad()) {
+  if (allowInsertSliceLowering && packOp.isLikePad()) {
```

Comment:
Give where this check is used ... I feels like the name is a bit misleading. Shouldn't this be more like ... `lowerPad`?

Reply:
Maybe something like `lowerPadLikeWithInsertSlice` and `lowerUnpadLikeWithExtractSlice`?